File: src/main/java/demo/bfims/Entities/Inventory/Journal.java
package demo.bfims.Entities.Inventory;

import demo.bfims.Enums.ItemType;
import demo.bfims.Enums.PublicationItemType;
import jakarta.persistence.*;

import java.util.Date;

@Entity
public class Journal extends PublicationItem {
    @Temporal(TemporalType.DATE)
    private Date issueDate;
    private int issueNumber;
    private String issueName;
    private String volume;

    public Journal() {
        this.setPublicationItemType(PublicationItemType.JOURNAL);
    }

    public int getIssueNumber() {
        return issueNumber;
    }

    public void setIssueNumber(int issueNumber) {
        this.issueNumber = issueNumber;
    }

    public String getIssueName() {
        return issueName;
    }

    public void setIssueName(String issueName) {
        this.issueName = issueName;
    }

    public String getVolume() {
        return volume;
    }

    public void setVolume(String volume) {
        this.volume = volume;
    }

    public Date getIssueDate() {
        return issueDate;
    }

    public void setIssueDate(Date issueDate) {
        this.issueDate = issueDate;
    }


    @Override
    public String toString() {
        return "Journal{" +
                "issueDate=" + issueDate +
                ", issueNumber=" + issueNumber +
                ", issueName='" + issueName + '\'' +
                ", volume='" + volume + '\'' +
                ", publication=" + publication +
                "} " + super.toString();
    }
}

File: frontend/src/app/DTOs/Inventory/Journal.ts
import {PublicationItem} from './PublicationItem';
import {PublicationItemType} from '../../Enums/PublicationItemType';

export class Journal extends PublicationItem {
  public issueDate: Date;
  public issueNumber: string;
  public issueName: string;
  public volume: string;

  constructor(...data: any) {
    data = data[0];
    super(data);
    if (data) {
      this.issueNumber = data["issueNumber"] || null;
      this.issueName = data["issueName"] || null;
      this.issueDate = data["issueDate"] || null;
      this.volume = data["volume"] || null;
    }
    this.publicationItemType = PublicationItemType.JOURNAL;
  }


}
